### PR TITLE
Add Disable siren

### LINKED ISSUE #1375: [Request]: Turn Siren off/on Option for Emergency Vehicles
### Problem

In „Menyoo“, there’s an option to (de)activate the siren of an emergency vehicle. I wish it for this menu. For now, I need to turn on the siren and go under water. Lights still on, but the vehicle is broken. Repairing the vehicle and I can drive around without sirens and activated lights. XD 

### Solution

In the vehicle controller, could be an option to turn off emergency lights.

### Reason

This is a cool and usefull feature.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by CypherMorgan**
closes #1375

### CODE REVIEW

**Comment by TheGreenBandit** [comment]
Please change to the command format. Also the way u structured needs editing

File: src/views/vehicle/view_vehicle.cpp [hunk 1]
```diff
@@ -93,6 +93,8 @@ namespace big
 			components::command_checkbox<"driveunder">();
 			components::command_checkbox<"keeponground">();
 
+			ImGui::Checkbox("Disable Siren", &g.vehcile.enable_siren_mute);
```
Comment (Rxann) [inline src/views/vehicle/view_vehicle.cpp]: vehicle misspell?

File: src/core/globals.hpp [hunk 2]
```diff
@@ -644,8 +644,9 @@ namespace big
...
+			NLOHMANN_DEFINE_TYPE_INTRUSIVE(vehicle, speedo_meter, fly, rainbow_paint, speed_unit, god_mode, proof_bullet, proof_fire, proof_collision, proof_melee, proof_explosion, proof_steam, proof_water, proof_mask, auto_drive_destination, auto_drive_style, auto_drive_speed, auto_turn_signals, boost_behavior, drive_on_water, horn_boost, instant_brake, block_homing, seatbelt, turn_signals, vehicle_jump, keep_vehicle_repaired, no_water_collision, disable_engine_auto_start, change_engine_state_immediately, vehinvisibility, localveh_visibility, localped_visibility, keep_on_ground, no_collision, unlimited_weapons, enable_siren_mute)
```
Comment (Rxann) [inline src/core/globals.hpp]: I would name this siren_mute personally, but it's up to you

Comment (CypherMorgan) [reply src/core/globals.hpp]: Changed..

File: src/views/vehicle/view_vehicle.cpp (hunk 1 above)
Comment (CypherMorgan) [reply src/views/vehicle/view_vehicle.cpp]: My bad, Fixed that too

**Comment by CypherMorgan** [comment]
> Please change to the command format. Also the way u structured needs editing

Hey, Kindly Take a look, First PR on Git so might be something wrong..

**Review by Rxann — COMMENTED** [review]
Looks good to me, but u def need someone better than me to look at it. I am also not at a place where I can access my pc, so all of this has been done from my phone 🥲

**Comment by CypherMorgan** [comment]
> Looks good to me, but u def need someone better than me to look at it. I am also not at a place where I can access my pc, so all of this has been done from my phone 🥲

Ok no issues!

File: src/backend/looped/vehicle/disable_siren.cpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+namespace big
+{
+	class siren_mute : looped_command
+	{
+		using looped_command::looped_command;
+
+		virtual void on_tick() override
+		{
+			Vehicle vehicle = self::veh;
+			bool toggle     = true; // Enable muted siren state
```
Comment (tupoy-ya) [inline src/backend/looped/vehicle/disable_siren.cpp]: Why make it a variable if you never change it?

File: src/backend/looped/vehicle/disable_siren.cpp [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+			if (ENTITY::DOES_ENTITY_EXIST(vehicle))
+				{
+					VEHICLE::SET_VEHICLE_HAS_MUTED_SIRENS(vehicle, toggle);
+				}
+		}
+
+		virtual void on_disable() override
+		{
+			Vehicle vehicle = self::veh;
+			bool toggle     = false; // Enable muted siren state
```
Comment (tupoy-ya) [inline src/backend/looped/vehicle/disable_siren.cpp]: You are disabling it here, but the comment says the opposite.

File: src/core/globals.hpp [hunk 5]
```diff
@@ -644,8 +644,9 @@ namespace big
 			bool keep_on_ground                         = false;
 			bool no_collision                           = false;
 			bool unlimited_weapons                      = false;
+			bool siren_mute				    = false;
```
Comment (tupoy-ya) [inline src/core/globals.hpp]: The entire section above uses spaces and you used tabs, you should probably run `clang-format`.

Or maybe it's just some new github UI bug, i don't now.

**Review by tupoy-ya — CHANGES_REQUESTED** [review]
Personally i would wait until [lua](https://github.com/YimMenu/YimMenu/pull/1334) gets merged and make it a lua script instead.

File: src/core/globals.hpp (hunk 5 above)
Comment (CypherMorgan) [reply src/core/globals.hpp]: Its a bug, if i use spaces it just puts it way ahead in github PR in my case

File: src/backend/looped/vehicle/disable_siren.cpp (hunk 4 above)
Comment (CypherMorgan) [reply src/backend/looped/vehicle/disable_siren.cpp]: Ah, ill fix it once im back on my pc

Comment (CypherMorgan) [reply src/backend/looped/vehicle/disable_siren.cpp]: Changed..

File: src/backend/looped/vehicle/disable_siren.cpp [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+namespace big
+{
+	class siren_mute : looped_command
+	{
+		using looped_command::looped_command;
+
+		virtual void on_tick() override
+		{
+			if (ENTITY::DOES_ENTITY_EXIST(self::veh))
+				{
```
Comment (tupoy-ya) [inline src/backend/looped/vehicle/disable_siren.cpp]: This tab after `if` is weird.

File: src/backend/looped/vehicle/disable_siren.cpp [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+		virtual void on_disable() override
+		{
+			if (ENTITY::DOES_ENTITY_EXIST(self::veh))
+				{
+					VEHICLE::SET_VEHICLE_HAS_MUTED_SIRENS(self::veh, false); // Disables muted siren state
+				}
+		}
+	};
+
+	siren_mute g_siren_mute("mutesiren", "Disable Siren", "Disables the siren sound of Emergency vehicles.",
```
Comment (tupoy-ya) [inline src/backend/looped/vehicle/disable_siren.cpp]: Maybe name it "Mute Siren" as well.

**Review by tupoy-ya — DISMISSED** [review]
I did not test this, but it looks fine.

File: src/backend/looped/vehicle/disable_siren.cpp (hunk 6 above)
Comment (CypherMorgan) [reply src/backend/looped/vehicle/disable_siren.cpp]: There's some real issue with this Git UI, There's not even a Tab there in my local repo. I'll try fix that

File: src/backend/looped/vehicle/disable_siren.cpp (hunk 7 above)
Comment (CypherMorgan) [reply src/backend/looped/vehicle/disable_siren.cpp]: Done

File: src/backend/looped/vehicle/disable_siren.cpp (hunk 6 above)
Comment (CypherMorgan) [reply src/backend/looped/vehicle/disable_siren.cpp]: I tried lemme know if its still there in ur pov as in my pov it just changed (like it was before).

Comment (tupoy-ya) [reply src/backend/looped/vehicle/disable_siren.cpp]: Seems fixed for me.

File: src/backend/looped/vehicle/disable_siren.cpp [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+		virtual void on_disable() override
+		{
+			if (ENTITY::DOES_ENTITY_EXIST(self::veh))
+			{
+				VEHICLE::SET_VEHICLE_HAS_MUTED_SIRENS(self::veh, false); // Disables muted siren state
+			}
+		}
+	};
+
+	siren_mute g_siren_mute("mutesiren", "Mute Siren", "Disables the siren sound of Emergency vehicles.",
```
Comment (tupoy-ya) [inline src/backend/looped/vehicle/disable_siren.cpp]: There is a dot in the end of the description, other commands don't have it for some reason.

Sorry for not saying it all in one review i'm blind.

File: src/backend/looped/vehicle/disable_siren.cpp (hunk 6 above)
Comment (CypherMorgan) [reply src/backend/looped/vehicle/disable_siren.cpp]: Okay, Thanks.

File: src/backend/looped/vehicle/disable_siren.cpp (hunk 8 above)
Comment (CypherMorgan) [reply src/backend/looped/vehicle/disable_siren.cpp]: Ah, i added to end the sentence, my bad.

Comment (CypherMorgan) [reply src/backend/looped/vehicle/disable_siren.cpp]: Done

**Review by tupoy-ya — APPROVED** [review]
Looks fine to me.

**Comment by CypherMorgan** [comment]
Alright.

**Review by Yimura — APPROVED** [review]

**Pull request merged**